group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -2191,11 +2191,7 @@ public void testStaticMemberGetsBackAssignmentUponRejoin() {
             .setMemberEpoch(-2)
             .build();
 
-        List<Record> expectedRecords = Collections.singletonList(
-            RecordHelpers.newCurrentAssignmentRecord(groupId, member2UpdatedEpoch)
-        );
-
-        assertEquals(result.records(), expectedRecords);
+        assertRecordEquals(result.records().get(0), RecordHelpers.newCurrentAssignmentRecord(groupId, member2UpdatedEpoch));
```

Comment:
this removes the assertion that there is only one record in `result.records()`. We can have more than one record in `result.records()` and earlier, it would have failed but now it will not.

Reply:
That's a good point. I mean for these tests, it is expected to have only 1 record, but in the future if the consumer group heartbeat logic changes, then we could have multiple records which would not get caught with these tests. I have added an assertion for the record size which would need to be tweaked for the future cases.